### DISCUSSION TITLE
Has anyone had any luck getting Sonic 64 to run yet?

### DISCUSSION THREAD

**Discussion by MC6809E089** [frntc/RAD#1] [General]
Hiya,

Firstly, many, many thanks for sharing your awesome work! I am really enjoying playing with the RAD, Sidekick and the SIDKicks I've built. 

I am wondering if you, or anyone has had success in getting Sonic 64 to run? The way I am trying at the moment is to load and start the BluREU Crest demo, and then reset to drop to back to basic. I can then get Sonic to load all the way to the title screen but it locks up and crashes if I try to start the game, or if the level demo starts. I am wondering if anyone knows a better way to try get it loading?

Thanks very much once again, have a great evening! :)

**Comment by frntc** [comment]
Hey,

in the main menu simply select REU emulation with 256kb or 512kb (or more, but not used), go to basic by pressing 'X' and then load from disk. That should work!?

**Reply by MC6809E089** [reply]
Thanks very much! I'll give that a try in a little while. :)

**Reply by MC6809E089** [reply]
Hi again, 

I tried what you suggested but unfortunately the game still crashes in the same manner, so I don't believe it is failing due to the way I was launching it previously. It still crashes just just past the title page, and a few seconds into the demo, or the game itself. 

Would you know if anyone else has tried the game and had better luck than me? If it does work okay for other people, then I can be sure it is me that did something wrong (It's usually something I did!). :)

Here's some more info, in case it is useful for any further development or other people trying the same thing...

I don't think it is my soldering (though always a possibility) because the other REU demos seem to work just fine.
I don't think the Pi1541 I am loading the game from is causing the issue. All other software I've tried loads and runs fine from the Pi drive. The game has the option of using a fastloader, or a compatible one but it crashes in the same way regardless of the loader selected.
I tried powering the RAD both externally via USB, and internally from the C64. It made no difference, the game still crashed in the same place, so I don't think the issue is power related.
I am using a Pi Zero 2 and the smaller version of the RAD, the logic chips for the RAD were all sourced from Mouser, and are exactly the same versions as those listed on the BOM, so I don't think I have bad chips. Sonic is the only thing I've tried so far which doesn't work properly for me, and I'd have expected other stuff to crash if there was something wrong with the RAD cart itself.

Any further help or ideas you might have would be very much appreciated but I do realise it's your free time, and I really don't expect you to keep giving support to random folks on the internet like me! I will keep my fingers crossed this piques your interest enough to investigate, 

Thanks again and have a great evening! :)

**Comment by frntc** [comment]
So far noone reported problems with Sonic. Have you tried a REU-test program such as: https://www.retro-programming.de/download/reu-checker/ ? Let's first see if this runs reliably. One possible cause is that you need to adjust the timings a bit, some reported their settings on forum64.de (there are discussion threads for RAD).

**Reply by MC6809E089** [reply]
Thank you very much! I will take a look at those properly later once I finish work for the day. I do wonder if it's timing related, as I had the RAD plugged into a sizty-clone board...

**Reply by frntc** [reply]
It's could be possible that timing adjustment are necessary for the sixtyclone-board (I don't have one, so I can't test myself)

**Reply by MC6809E089** [reply]
Hiya,

Sorry for not replying sooner, I've only just had the time to test things this evening. I tried the REU test program and that passed fine, so I tried changing the timings but that didn't seem to make any difference, and then I remembered something on forum64 about PLAs, and I realised the sixty-clone (which is a long board) has an XCPLA in it and not a real one.

I wondered if that might be the reason for Sonic64 failing, so I went up into the loft and pulled down an original shortboard with a rev B SuperPLA in it, and both the RAD, the test program, the demos, and also Sonic work fine on that! 

Then I swapped the original shortboard for a SixtyClone shortboard (don't ask, I have too many C64 mainboards I think! :) ), and that works fine too as it has an original Commodore Rev A SuperPLA + colour RAM in it.

I should have thought to try a different mainboard sooner but this is the first time I've had anything not work on that SixtyClone longboard, it has always worked perfectly until now. It's a good thing you pointed me towards the forum, or I'd probably still be scratching my head and wondering what was wrong. :)

I'll try swapping out the XCPLA for one I have from from iComp (hopefully tomorrow if I have time) to see if that makes any difference. I don't think I have an original longboard PLA anymore but I'll try and get one ordered to see if that changes things too (I suspect it will fix everything). I'll report back once I've tried them, so the information is here in case other folks need it.

Thanks very much once again for all of your help! :)

**Reply by frntc** [reply]
Let me know how it goes! I tested on various boards and with different PLAs, but it might always happen that some combination (or replica board ;-)) requires different timings.

**Comment by MC6809E089** [comment]
Well, after a lot of trying and testing things (and a fair bit of swearing), I've finally worked out what was causing the issue...

It wasn't the PLA. It also wasn't the CPU, or the VIC-II, or the SIDKick, or any of the other logic ics on the 250466 SixtyClone. The cause was... the replacement SRAM module. 

For some reason (I guess timing related? but it could be capacitance. load/line driving, or something else), the SRAM modules I have do not play nice with the RAD, on a 250466 mainboard. I've had this particular SRAM module plugged into the board since I built it (2 years or so) and this is the first time I've had any kind of problem. The module I was using is an open source one which I got the designs for from here... https://github.com/jamarju/c64-sram

The same module and RAD work fine on my SixtyClone 250469 shortboard, and also on an original shortboard, but not on the 250466. No amount of me messing about with the timings seemed to make it more stable on the 250466, it just crashed consistently every time. I have 2 spare SRAM modules of the same type, so I tried those, in case the first was bad somehow, but they all behaved the same.

I swapped out the SRAM for a pair of 41464s, and have left Sonic running in the background while I work today. It's been going for about 7 hours so far, and it hasn't crashed at all, though I might go insane if I have to listen to the tune for much longer... So I think that's a win. 

If anyone else is using the same SRAM module and it's working fine, or if anyone can recommend a different SRAM replacement, that's okay in a 250466, please let me know as I would like to have a backup plan for avoiding old RAM and VSP bugs.

Thanks very much for all your help, have a great weekend! :)
